src/algebra/squarefree.lean

```diff
@@ -85,6 +85,42 @@ end
 
 end multiplicity
 
+lemma eq_zero_or_no_irreducibles_and_squarefree_iff_irreducible_sq_not_dvd
+  (r : R) [comm_monoid_with_zero R] [wf_dvd_monoid R] :
```

Comment:
Why this order of variables? Usually we put typeclass assumption first, and `(r : R)` last.

Reply:
My mistake, I will amend!